launcher/internal/signaturediscovery/client.go

```diff
@@ -8,31 +8,38 @@ import (
 	"github.com/containerd/containerd"
 	"github.com/containerd/containerd/content"
 	"github.com/containerd/containerd/images"
+	"github.com/google/go-tpm-tools/launcher/auth"
 	"github.com/google/go-tpm-tools/verifier/oci"
 	"github.com/google/go-tpm-tools/verifier/oci/cosign"
 	v1 "github.com/opencontainers/image-spec/specs-go/v1"
+	"golang.org/x/oauth2"
 )
 
 const signatureTagSuffix = "sig"
 
+type (
+	oauth2TokenFetcher func(context.Context) (oauth2.Token, error)
```

Comment:
Instead of having the logic in the signature client, can we move it to auth.go package? So the resolver can handle the token fetching itself

Reply:
Done.